service/db.go

```diff
@@ -626,3 +651,62 @@ func isEmailTaken(prospectiveEmail string) (bool, error) {
 
 	return false, nil
 }
+
+// getEmailTokenRow looks up existing token from user_svc.email_tokens table.
+// If token exists, the rows information are returned in a tokenEmailRow struct.
+// If token does not exist, return error.
+func getEmailTokenRow(token string) (*tokenEmailRow, error) {
+	if token == "" {
+		return nil, status.Error(codes.InvalidArgument, authconst.ErrEmptyToken.Error())
```

Comment:
I thought this was what we agreed on earlier about sending specific grpc error codes.

Reply:
I was not sure where the change was going to happen. This code review gave me a lot to think about.

Reply:
What do you propose?

I don't know how else you would achieve retrieving specific grpc error codes without doing extra work. This isn't really extra work (other than refactoring), we are just specifying the specific type of grpc error code earlier on before returning the error.

Reply:
Please return a non grpc error. This is a DAO. I want to separate the concern accessing the DB. Making `db.go` return a non-grpc error means `db.go` can be extended to be used by other source codes because it is using a generic error.

Reply:
i did the same mistake at work, and i got lectured.